### PR TITLE
gcd/lcm test from internal repo with questionable bits extracted

### PR DESCRIPTION
**Pull request by CaseyCarter**
Partially addresses VSO-936680

### CODE REVIEW

File: tests/std/tests/P0295R0_gcd_lcm/test.cpp [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+#include <limits.h>
+#include <numeric>
+
+using std::gcd;
+using std::lcm;
+
+#define STATIC_ASSERT(...) static_assert(__VA_ARGS__, #__VA_ARGS__)
+
+int main() {} // COMPILE-ONLY
```
Comment (StephanTLavavej) [inline tests/std/tests/P0295R0_gcd_lcm/test.cpp:12]: This test should be named `test.compile.pass.cpp` so the GitHub harness doesn't bother trying to run it.

**Review by StephanTLavavej — APPROVED** [review]

**Comment by StephanTLavavej** [comment]
I'm mirroring this to the MSVC-internal repo - please notify me if any further changes are pushed.

**Review by barcharcraz — APPROVED** [review]
do we prefer the test.compile.pass.cpp format over just test.cpp now?

**Comment by CaseyCarter** [comment]
> do we prefer the test.compile.pass.cpp format over just test.cpp now?

Yes we do. The GitHub test runner recognizes the suffix `.compile.pass.cpp` and it won't attempt to link and run such test cases.

**Comment by StephanTLavavej** [comment]
One note about `test.compile.pass.cpp` - such tests must still be prepared to be linked and ran with no effect, since the internal test harness doesn't know to skip those steps. (It could presumably be enlightened, but teaching that ancient Perl machinery anything is quite difficult.)

**Comment by StephanTLavavej** [comment]
Thanks for being the Greatest Casey Denominator! :joy_cat: :arrow_up_small: :tada:

**Pull request merged**